### PR TITLE
Navigation: Browse Mode: Fix misaligned ellipsis

### PR DESCRIPTION
**Pull request by scruffian**
## What?
Aligns the ellipsis menus in Browse mode.

## Why?
So it looks better.

## How?
Add some negative margins to the list view so it can use more of the screen canvas. Unfortunately using a grid value isn't possible as the value we need is between grid values.

## Testing Instructions
1. Open the Navigation section of Browse Mode in the Site Editor
2. Select a navigation
3. Check whether the ellipsis menu is aligned with the one in the header.

### Testing Instructions for Keyboard
N/A. This is a visual change

## Screenshots or screencast 
Screenshot 2023-06-16 at 14 43 51

### CODE REVIEW

File: packages/edit-site/src/components/sidebar-navigation-screen-template-part/template-part-navigation-menu.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ export default function TemplatePartNavigationMenu( { id } ) {
 				size="12"
 				upperCase={ true }
 			>
-				{ title || __( 'Navigation' ) }
```
Comment (MaggieCabrera) [inline packages/edit-site/src/components/sidebar-navigation-screen-template-part/template-part-navigation-menu.js]: Sneaky unrelated change?

**Comment by MaggieCabrera** [comment]
I see the problem, but I'm not a fan of the "magic" numbers for pushing things around. Looks like the container for the ellipsis menu in the list view has a width of 24px, while the one next to the title is 36px. We could solve this by removing the padding on the ellipsis menu on the title.

Something like `.sidebar-navigation__more-menu .components-button.has-icon { padding:0;}` should do the trick. It might need tweaking the min-width value to 24px too (that must be a sass variable somewhere)

**Comment by scruffian** [comment]
> We could solve this by removing the padding on the ellipsis menu on the title.

I'm reluctant to do this as I assume that the ellipsis position in the title area is the correct one. My preference is to make the lower section wider to also give more space for longer navigation items, but I take your point that magic values aren't good. Closing this for now.

**Comment by MaggieCabrera** [comment]
There's a padding that makes the lower section narrower (in `.edit-site-sidebar-navigation-screen__content`), maybe we wanna play with that. @jasmussen you said there were some visual polishing coming up for browse mode, is this included?

**Comment by jasmussen** [comment]
I think @jameskoster has driven some of those design efforts, and might be better equipped to answer.

**Comment by jameskoster** [comment]
Some metrics may change for 6.3, but I don't sense a great deal of urgency around that, so they may not. 

> I'm reluctant to do this as I assume that the ellipsis position in the title area is the correct one

I think this is a good assumption. And so it makes most sense for the Navigation elements to align themselves with what already exists in the sidebar, rather than the other way around.

`<ItemGroup>`s have a negative margin applied in order to align their children. You can see this in effect on the main Design panel, Pages, and Templates lists too. I don't know if it's good practise, but could the navigation menu be wrapped in an `<ItemGroup>`? That gets us here:

Screenshot 2023-06-16 at 16 34 15

The remaining small misalignment is due to the ellipsis buttons using different size variants. The panel header uses the default size (36px) while the navigation menu item uses the small variant (24px). Could some logic be added to the navigation menu items to use the default size variant in the sidebar? That gets us to perfect alignment in a way that avoids magic numbers and should be backwards compatible:

Screenshot 2023-06-16 at 16 40 22
